frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import { post } from '../../lib/k8s/apiProxy';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton } from '../common';
+
+export default function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceNameValid, setNamespaceNameValid] = useState(false);
+  const [nameHelperMessage, setNameHelperMessage] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    const clusterData = getCluster();
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+      },
+    };
+
+    const newNamespaceName = newNamespaceData.metadata.name;
+
+    async function namespaceRequest() {
+      try {
+        return await post('/api/v1/namespaces', newNamespaceData, true, {
+          cluster: clusterData || '',
+        });
+      } catch (error) {
+        console.error('Error creating namespace', error);
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        if (errorMessage.includes('already exists')) {
+          setNamespaceDialogOpen(true);
+          setNamespaceNameValid(false);
+          setNameHelperMessage(t('translation|Namespace name already exists.'));
```

Comment:
This text sounds a bit odd. Maybe just "Namespace already exists" or "A namespace with this name already exists"